pkg/operator/controller/dns_status.go

```diff
@@ -100,23 +100,22 @@ func computeDNSDegradedCondition(oldCondition *operatorv1.OperatorCondition, clu
 		want := dnsDaemonset.Status.DesiredNumberScheduled
 		have := dnsDaemonset.Status.NumberAvailable
 		numberUnavailable := want - have
-		maxUnavailableIntStr := intstr.FromInt(1)
-		if dnsDaemonset.Spec.UpdateStrategy.RollingUpdate != nil && dnsDaemonset.Spec.UpdateStrategy.RollingUpdate.MaxUnavailable != nil {
-			maxUnavailableIntStr = *dnsDaemonset.Spec.UpdateStrategy.RollingUpdate.MaxUnavailable
-		}
+		maxUnavailableIntStr := intstr.FromString("10%")
 		maxUnavailable, intstrErr := intstr.GetScaledValueFromIntOrPercent(&maxUnavailableIntStr, int(want), true)
+
 		switch {
 		case want == 0:
 			status = operatorv1.ConditionTrue
 			degradedReasons = append(degradedReasons, "NoDNSPodsDesired")
 			messages = append(messages, "No DNS pods are desired; this could mean all nodes are tainted or unschedulable.")
+		case intstrErr != nil:
+			// This should not happen, but is included just to safeguard against future changes.
+			degradedReasons = append(degradedReasons, "InvalidDNSMaxUnavailable")
+			messages = append(messages, fmt.Sprintf("The DNS daemonset has an invalid MaxUnavailable value: %v", intstrErr))
```

Comment:
Was there a reason to move this case up?

Reply:
Fixed, forgot to push.